src/track/cueinfo.cpp

```diff
@@ -4,6 +4,33 @@
 
 namespace mixxx {
 
+namespace {
+
+inline void assertEndPosition(
```

Comment:
`inline` will do nothing here IIUC. `inline` is only a directive that is needed to avoid errors during linking when functions are implemented in headers and included in multiple translation units. This is a legacy thing from C. in C++ everything implemented in headers is implicitly marked inline anyways. If you don't want the compiler to generate the function in case its not needed (such as when compiling without DEBUG_ASSERTs), you should instead mark the function `static` to tell the compiler its only needed in this translation unit and when not needed, it will inline and completely remove the noop body. 
```suggestion

static void assertEndPosition(
```

Reply:
It is already in the anonymous namespace that means the same as static. 

It depends of the function length if the compiler inlines it or not. In this case the inline keyword forces the compiler to inline the code even if the function is long. See
https://godbolt.org/z/99MsedsYP

My original intention is that everything is optimized away in case of Release builds without adding any #ifdef clutter. 
This will probably happen anyway in this case so I will remove the inline keyword.

Reply:
I forgot about the implications of the anonymous namespace. Then static is not needed.
Thank you also for the example. While obviously making a difference in your demo, the standard does not dictate that the `inline` has any influence on the optimizer. See [this paragraph from cppreference](https://en.cppreference.com/w/cpp/language/inline):

> The original intent of the inline keyword was to serve as an indicator to the optimizer that inline substitution of a function is preferred over function call, that is, instead of executing the function call CPU instruction to transfer control to the function body, a copy of the function body is executed without generating the call. This avoids overhead created by the function call (passing the arguments and retrieving the result) but it may result in a larger executable as the code for the function has to be repeated multiple times.
> 
> Since this meaning of the keyword inline is non-binding, compilers are free to use inline substitution for any function that's not marked inline, and are free to generate function calls to any function marked inline. Those optimization choices do not change the rules regarding multiple definitions and shared statics listed above. 


> My original intention is that everything is optimized away in case of Release builds without adding any #ifdef clutter.

Yes, I agree with that goal. However, since the DEBUG_ASSERTs compile to noops, anyways, the entire switch is a noop, and thus is the function. Since the compiler knows the definition while optimizing the callsites (since the definition is static), it can infer that the function is a noop too and thus the call will vanish (even at `-O1` or `-Os`). 

Thank you for removing the inline keyword, the discussion is resolved. :+1: